src/main/java/teammates/common/util/LogHelper.java

```diff
@@ -0,0 +1,196 @@
+package teammates.common.util;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import teammates.common.exception.InvalidParametersException;
+
+import com.google.appengine.api.log.AppLogLine;
+import com.google.appengine.api.log.LogQuery;
+import com.google.appengine.api.log.LogServiceFactory;
+import com.google.appengine.api.log.RequestLogs;
+import com.google.appengine.api.log.LogService.LogLevel;
+
+/**
+ * An utility to fetch logs from GAE server.
+ */
+public class LogHelper {
+    /**
+     * Maximum number of versions to query.
+     * The current value will include the current version and its 5 preceding versions.
+     */
+    private static final int MAX_VERSIONS_TO_QUERY = 6;
+    
+    /**
+     * A flag to decide whether to include application logs in result or not.
+     */
+    private static final boolean INCLUDE_APP_LOG = true;
+    
+    /**
+     * Affects the internal strategy to get logs. It doesn't affect the result.
+     */
+    private static final int BATCH_SIZE = 1000;
+    private static final LogLevel MIN_LOG_LEVEL = LogLevel.INFO;
+    public static final int SEARCH_TIME_INCREMENT = 2 * 60 * 60 * 1000;  // two hours in milliseconds
+    
+    private LogQuery query;
+    private Long endTime;
+    private List<String> versionList;
+    
+    public LogHelper() {
+        query = LogQuery.Builder.withDefaults();
+        query.includeAppLogs(INCLUDE_APP_LOG);
+        query.batchSize(BATCH_SIZE);
+        query.minLogLevel(MIN_LOG_LEVEL);
+    }
+    
+    /**
+     * Sets values for query.
+     * If versionsToQuery is null or empty, the current version with its 5 preceding versions will be used instead.
+     * 
+     * @param versionsToQuery 
+     * @param startTime
+     * @param endTime
+     */
+    public void setQuery(List<String> versionsToQuery, Long startTime, Long endTime) {
+        setTimePeriodForQuery(startTime, endTime);
+        setEndTime(endTime);
+        versionList = getVersionIdsForQuery(versionsToQuery);
+        query.majorVersionIds(versionList);
+    }
+    
+    /**
+     * Sets time period to search for query.
+     * @param startTime
+     * @param endTime
+     */
+    private void setTimePeriodForQuery(Long startTime, Long endTime) {
+        if (startTime != null) {
+            query.startTimeMillis(startTime);
+        }
+        if (endTime != null) {
+            query.endTimeMillis(endTime);
+        }
+    }
```

Comment:
1. Why this method checks for null but `setEndTime` method below doesn't?
2. Shouldn't this method call the `setEndTime` method?